src/v/datalake/translation/partition_translator.cc

```diff
@@ -103,7 +105,10 @@ partition_translator::partition_translator(
   , _features(features)
   , _cloud_io(cloud_io)
   , _schema_mgr(schema_mgr)
+  // TODO: type resolver and record translator should be constructed based on
+  // topic configs.
   , _type_resolver(type_resolver)
+  , _record_translator(std::make_unique<default_translator>())
```

Comment:
NOTE: this is where we should plug in the Iceberg mode topic config:

- `key_value`: `binary_type_resolver` and `key_value_translator`
- `schema_id_and_prefix`: `record_schema_resolver` and `structured_data_translator`